packages/volto/src/helpers/Robots/Robots.js

```diff
@@ -15,41 +17,23 @@ import { addHeadersFactory } from '@plone/volto/helpers/Proxy/Proxy';
  */
 export const generateRobots = (req) =>
   new Promise((resolve) => {
-    const internalUrl =
-      config.settings.internalApiPath ?? config.settings.apiPath;
-    const request = superagent.get(`${internalUrl}/robots.txt`);
-    request.set('Accept', 'text/plain');
+    const request = superagent.get(formatUrl('@site'));
+    request.set('Accept', 'application/json');
     const authToken = req.universalCookies.get('auth_token');
     if (authToken) {
       request.set('Authorization', `Bearer ${authToken}`);
     }
     request.use(addHeadersFactory(req));
-    request.end((error, { text }) => {
+    request.end((error, { text, body }) => {
       if (error) {
         resolve(text || error);
       } else {
-        // It appears that express does not take the x-forwarded headers into
-        // consideration, so we do it ourselves.
-        const {
-          'x-forwarded-proto': forwardedProto,
-          'x-forwarded-host': forwardedHost,
-          'x-forwarded-port': forwardedPort,
-        } = req.headers;
-        const proto = forwardedProto ?? req.protocol;
-        const host = forwardedHost ?? req.get('Host');
-        const portNum = forwardedPort ?? req.get('Port');
-        const port =
-          (proto === 'https' && '' + portNum === '443') ||
-          (proto === 'http' && '' + portNum === '80')
-            ? ''
-            : `:${portNum}`;
-        // Plone has probably returned the sitemap link with the internal url.
-        // If so, let's replace it with the current one.
-        const url = `${proto}://${host}${port}`;
-        text = text.replace(internalUrl, url);
-        // Replace the sitemap with the sitemap index.
-        text = text.replace('sitemap.xml.gz', 'sitemap-index.xml');
-        resolve(text);
+        resolve(
+          body['plone.robots_txt'].replace(
+            '{portal_url}',
+            config.settings.publicURL,
+          ),
```

Comment:
This replacement should be documented. Also, I think it would be good to keep the replacement of the internal URL with the public URL, in case the robots.txt from the backend includes an (internal) sitemap URL.

Reply:
Where should this be documented? I'd suggest:

- A new page under [User Manual](https://6.docs.plone.org/volto/user-manual/index.html), called "Site Configuration" with the filename `site-configuration.md`.
- [Settings reference guide](https://6.docs.plone.org/volto/configuration/settings-reference.html)